Neos.Media.Browser/Classes/Controller/AssetController.php

```diff
@@ -294,6 +299,9 @@ public function newAction()
      */
     public function replaceAssetResourceAction(Asset $asset)
     {
+        $assetSourceIdentifier = $asset->getAssetSourceIdentifier();
+        $this->applyActiveAssetSourceToBrowserState($assetSourceIdentifier);
```

Comment:
this is required to be sure, that the current asset source in the browser state, is the one attached with the current asset.